framework/db/BaseActiveRecord.php

```diff
@@ -418,7 +418,13 @@ public function getRelatedRecords()
      */
     public function hasAttribute($name)
     {
-        return isset($this->_attributes[$name]) || in_array($name, $this->attributes());
+        static $attributes = [];
+
+        if (!array_key_exists($name, $attributes)) {
+            $attributes[$name] = isset($this->_attributes[$name]) || in_array($name, $this->attributes());
+        }
+
+        return $attributes[$name];
```

Comment:
This might be a different matter though, because here we are calling BaseActiveRecord.attributes() instead of Model.attributes().

In a Model, it might make sense to override `attributes()` based on information in the current instance. But in ActiveRecord, `attributes()` is tied to the definition of the database backend, so I can't think of a reason to do so there. (You would use scenarios instead)